params/config.go

```diff
@@ -534,6 +536,14 @@ func (c *NodeConfig) Validate() error {
 		return err
 	}
 
+	// Whisper's data directory must be relative to the main data directory
+	// if EnableMailServer is true.
+	if c.WhisperConfig.Enabled && c.WhisperConfig.EnableMailServer {
+		if !strings.Contains(c.WhisperConfig.DataDir, c.DataDir) {
```

Comment:
Maybe it would be better to use something like [filepath.Rel](https://golang.org/src/path/filepath/path.go?s=7689:7740#L253) to check for this?

```suggestion
		if _, err = filepath.Rel(c.DataDir, c.WhisperConfig.DataDir); err != nil {
```

Reply:
I tried it but the problem with `Rel()` is that it tries to create a path always: https://play.golang.org/p/2VS3zuRfGSf. We can specify two totally different paths and it will try to figure out a relative path.

Reply:
Wouldn't it then be enough to test that `Rel()` doesn't return something that starts with `../`?

Reply:
Yes but what's the point of using `Rel()` then? I don't see any advantage of that. We explicitly want the Whisper dir path to start with the main data dir... And I have just realised I should use `strings.HasPrefix` instead of `strings.Contains` :D

Reply:
Yeah, that was one of the main reasons I was looking for an alternative. I guess if we call `Clean` on both paths and use `strings.HasPrefix`, that would do most of the same job as `Rel()`.

Reply:
I would not call `Clean` and move this responsibility on user to provide clean paths.